src/static/js/modules/util/web-storage-proxy.js

```diff
@@ -11,90 +11,107 @@
- *   Use non-persistent storage (true) or persistent storage (false).
+ * @param {object} storage (Optional)
+ *   Use non-persistent storage (sessionStorage)
+ *   or persistent storage (localStorage).
  * @returns {string} The value set in web storage.
  */
-function getItem( key, sessionOnly ) {
-  var storage = _sessionOrLocal( sessionOnly );
+function getItem( key, storage ) {
+  storage = _storageType( storage );
+
   return storage.getItem( key );
 }
 
 /**
  * Remove an item specified by the key.
  * @param {string} key The key for the value.
- * @param {boolean} sessionOnly (Optional)
- *   Use non-persistent storage (true) or persistent storage (false).
+ * @param {object} storage (Optional)
+ *   Use non-persistent storage (sessionStorage)
+ *   or persistent storage (localStorage).
  * @returns {boolean} Returns true if the item existed and it was
  *   removed. Returns false if the item didn't exist to begin with.
  */
-function removeItem( key, sessionOnly ) {
-  var storage = _sessionOrLocal( sessionOnly );
+function removeItem( key, storage ) {
+  storage = _storageType( storage );
   var returnVal = true;
-  if ( !storage.getItem( key ) ) returnVal = false;
-  if ( returnVal ) storage.removeItem( key );
+
+  if ( !storage.getItem( key ) ) {
+    returnVal = false;
+  }
+
+  if ( returnVal ) {
+    storage.removeItem( key );
+  }
 
   return returnVal;
 }
 
 /**
- * Internal function for setting default session type.
- * @param {boolean} sessionOnly
- *   Use non-persistent storage (true) or persistent storage (false).
- * @throws {Error} If parameter isn't a boolean.
+ * Set the default session type.
+ * @param {object} storage
+ *   Use non-persistent storage (sessionStorage)
+ *   or persistent storage (localStorage).
+ * @throws {Error} If parameter isn't a object.
  */
-function setSessionType( sessionOnly ) {
-  if ( typeof sessionOnly !== 'boolean' ) {
-    throw new Error( 'Setting requires a boolean value.' );
+function setStorage( storage ) {
+  if ( typeof storage !== 'object' ) {
+    throw new Error( 'Setting must be an object.' );
   }
-  _sessionOnly = sessionOnly;
+
+  _storage = storage;
 }
 
 /**
  * Internal function for whether to use local or session storage.
- * @param {boolean} sessionOnly
- *   Use non-persistent storage (true) or persistent storage (false).
+ * @param {object} storage
+ *   Use non-persistent storage (sessionStorage)
+ *   or persistent storage (localStorage).
  * @returns {object} A local storage or session storage instance.
  */
-function _sessionOrLocal( sessionOnly ) {
+/* istanbul ignore next */
+function _storageType( storage ) {
   // Use default setting if none is provided.
-  if ( typeof sessionOnly === 'undefined' ) sessionOnly = _sessionOnly;
-  var storage;
-  if ( sessionOnly ) storage = sessionStorage;
-  else storage = localStorage;
+  if ( typeof storage !== 'object' ) {
+    if ( typeof _storage === 'undefined' ) {
+      storage = window.sessionStorage;
+    } else {
```

Comment:
I don't think so. If `_storage` is undefined, we'd be overwriting `storage` as `undefined` after the inner `if {}`. Did you mean to write it as 

```
if ( typeof storage !== 'object' ) {
  if ( typeof _storage === 'undefined' ) {
    _storage = window.sessionStorage;
  }
  storage = _storage;
}
```

because I think that would work.